sdk/core/azure-core/tests/test_retry_policy.py

```diff
@@ -0,0 +1,157 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Tests for the retry policy."""
+try:
+    from io import BytesIO
+except ImportError:
+    from cStringIO import StringIO as BytesIO
+import pytest
+from azure.core.exceptions import AzureError
+from azure.core.pipeline.policies import (
+    RetryPolicy,
+    RetryMode,
+)
+from azure.core.pipeline import Pipeline, PipelineResponse
+from azure.core.pipeline.transport import (
+    HttpRequest,
+    HttpResponse,
+    HttpTransport,
+)
+
+def test_retry_code_class_variables():
+    retry_policy = RetryPolicy()
+    assert retry_policy._RETRY_CODES is not None
+    assert 408 in retry_policy._RETRY_CODES
+    assert 429 in retry_policy._RETRY_CODES
+    assert 501 not in retry_policy._RETRY_CODES
+
+def test_retry_types():
+    history = ["1", "2", "3"]
+    settings = {
+        'history': history,
+        'backoff': 1,
+        'max_backoff': 10
+    }
+    retry_policy = RetryPolicy()
+    backoff_time = retry_policy.get_backoff_time(settings)
+    assert backoff_time == 4
+
+    retry_policy = RetryPolicy(retry_mode=RetryMode.Fixed)
+    backoff_time = retry_policy.get_backoff_time(settings)
+    assert backoff_time == 1
+
+    retry_policy = RetryPolicy(retry_mode=RetryMode.Exponential)
+    backoff_time = retry_policy.get_backoff_time(settings)
+    assert backoff_time == 4
+
+@pytest.mark.parametrize("retry_after_input", [('0'), ('800'), ('1000'), ('1200')])
+def test_retry_after(retry_after_input):
+    retry_policy = RetryPolicy()
+    request = HttpRequest("GET", "https://bing.com")
+    response = HttpResponse(request, None)
+    response.headers["retry-after-ms"] = retry_after_input
+    pipeline_response = PipelineResponse(request, response, None)
+    retry_after = retry_policy.get_retry_after(pipeline_response)
+    seconds = float(retry_after_input)
+    assert retry_after == seconds/1000.0
+    response.headers.pop("retry-after-ms")
+    response.headers["Retry-After"] = retry_after_input
+    retry_after = retry_policy.get_retry_after(pipeline_response)
+    assert retry_after == float(retry_after_input)
+    response.headers["retry-after-ms"] = 500
```

Comment:
You should clarify with a comment that those three lines are the behavior when three values are provided, took me some time to get it :p